crates/bevy_ui/src/ui_node.rs

```diff
@@ -368,20 +369,38 @@ impl From<Color> for UiColor {
     }
 }
 
+/// The texture sheet index of the node.
+///
+/// Allows to customize the index of [`UiImage::TextureAtlas`]
+#[derive(Component, Default, Clone, Debug, Reflect, Deref, DerefMut)]
+#[reflect(Component, Default)]
+pub struct UiTextureIndex(pub usize);
+
 /// The image of the node
-#[derive(Component, Clone, Debug, Reflect, Deref, DerefMut)]
+#[derive(Component, Clone, Debug, Reflect)]
 #[reflect(Component, Default)]
-pub struct UiImage(pub Handle<Image>);
+pub enum UiImage {
+    /// Single texture
+    Image(Handle<Image>),
+    /// Texture Atlas
+    TextureAtlas(Handle<TextureAtlas>),
```

Comment:
Having the texture index part of this variant of the enum would be better. It doesn't make sense to have a `UiTextureIndex` without a `UiImage::TextureAtlas`

Reply:
That's indeed better, I don't know why I didn't think of that :o